src/content/docs/en/reference/configuration-reference.mdx

```diff
@@ -1635,3 +1635,220 @@ The example below displays a generic avatar as fallback content, then animates i
+  schema: z.object({
+    id: z.string(),
+    breed: z.string(),
+    temperament: z.array(z.string()),
+  }),
+});
+
+export const collections = { blog, dogs };
+```
+
+#### Querying and rendering with the Content Layer API
+
+The collection can be [queried in the same way as content collections](/en/guides/content-collections/#querying-collections):
+
+```ts
+// src/pages/index.astro
+import { getCollection, getEntry } from 'astro:content';
+
+// Get all entries from a collection.
+// Requires the name of the collection as an argument.
+const allBlogPosts = await getCollection('blog');
+
+// Get a single entry from a collection.
+// Requires the name of the collection and ID
+const labradorData = await getEntry('dogs', 'labrador-retriever');
+```
+
+Entries generated from Markdown, MDX, or Markdoc can be rendered directly to a page using the `render()` function.
+
+:::note
+The syntax for rendering collection entries is different from the current content collections syntax.
+:::
+
+```astro title="src/pages/[slug].astro"
+---
+import { getEntry, render } from 'astro:content';
+
+const post = await getEntry('blog', Astro.params.slug);
+
+const { Content, headings } = await render(entry);
+---
+
+<Content />
+```
+
+#### Creating a loader
+
+With the Content Layer API, you can build loaders to load or generate content from anywhere.
+
+For example, you can create a loader that fetches collection entries from a remote API.
+
+```ts
+// src/content/config.ts
+const countries = defineCollection({
+  loader: async () => {
+    const response = await fetch("https://restcountries.com/v3.1/all");
+    const data = await response.json();
+    // Must return an array of entries with an id property,
+    // or an object with IDs as keys and entries as values
+    return data.map((country) => ({
+      id: country.cca3,
+      ...country,
+    }));
+  },
+  // optionally add a schema
+  // schema: z.object...
+});
+
+export const collections = { countries };
+```
+
+For more advanced loading logic, you can define an object loader. This allows incremental updates and conditional loading while also giving full access to the data store. See the API in [the Content Layer API RFC](https://github.com/withastro/roadmap/blob/content-layer/proposals/0047-content-layer.md#loaders).
+
+#### Migrating an existing content collection to use the Content Layer API
+
+You can convert an existing content collection with Markdown, MDX, Markdoc, or JSON entries to use the Content Layer API.
+
+1. **Move the collection folder out of `src/content/`** (e.g. to `src/data/`). All collections located in the `src/content/` folder will use the existing Content Collections API.
+
+    **Do not move the existing `src/content/config.ts` file**. This file will define all collections, using either API.
+
+2. **Edit the collection definition**. Your updated collection requires a `loader`, and the option to select a collection `type` is no longer available.
+
+    ```diff
```

Comment:
```suggestion
    ```astro ins={3,8} del={7}
```

Should we use expressiveCode syntax instead of diff?

Reply:
Yes, we can! This came from the astro file, and it's not always as obvious when you're writing there. Thank you!